whois-commons/src/main/java/net/ripe/db/whois/common/apiKey/OAuthSession.java

```diff
@@ -79,27 +79,36 @@ public String toString() {
                 .add("keyId", keyId)
                 .add("email", email)
                 .add("uuid", uuid)
-                .add("scopes", scope)
+                .add("scope", scope)
                 .toString();
     }
 
    public static class ScopeFormatter {
 
         final String appName;
         final String scopeType;
+        final String scopeEnv;
```

Comment:
This is not needed right ? Can you rename this to whoisScopeFormatter and for environment check  just check hardcoded value as you don't need it anywhere else

Reply:
I need this to extract the environment from the key that the user is passing. I don't understand what you mean here.
The key could have been created in RC environment while this is deployed in PROD environment. We need to discard that key because we are not in the correct environment (there is a IT for that already)

Reply:
In addition, I am using this parser from whois-internal as well. So I need it there as well, and the logic here is the same for whois/whois-internal so we should keep it in one place for maintainability (we just need to change the parser in one place)

Reply:
In that case, can you use same scopeKey as envName and scopeType as ENVIRONMENT? You can also make it scopeType as enum with MNTNR and ENVIRONMENT instead of adding scopeEnv as new variable

Reply:
This will make it easy to extend multiple scope in future with scopeType for ex: for IP address level scope scopeType will be IP and scopeKey as ip address

Reply:
yes, I was thinking about that already. I think that is a good idea

Reply:
This is done now. Thanks for suggesting this change